### ISSUE TITLE
Investigate how Volley appears in Android Studio dependency manager

### ISSUE THREAD

**Issue by RobertBaruch** [google/volley#353]
As in #200 I am getting a NoClassDefFoundError with API 30 when attempting to follow the "[Send a simple request](https://developer.android.com/training/volley/simple)" lesson.

I think that fits this criterion mentioned in #200 : 

```
If you are just using Volley.newRequestQueue without providing an HttpStack, or if you are
using HurlStack or another stack which extends BaseHttpStack, the library should not attempt
to use Apache HTTP at runtime, and you should not need to request the library in your
AndroidManifest.xml.

If you are having a problem with Volley, and you are not using the Apache library or don't
believe you should be, then please let us know.
```

I am definitely not using the Apache library, and I don't believe I should be. I am definitely not providing an HttpStack, since the lesson specifies no such thing. I am not using HurlStack, although apparently calling newRequestQueue(context) sets up HurlStack for you. Note that if there is an additional dependency that Volley requires, it should be mentioned in the lesson, but it is not.

Here's the stack trace:

```
    java.lang.NoClassDefFoundError: Failed resolution of: Lorg/apache/http/ProtocolVersion;
        at com.android.volley.toolbox.HurlStack.performRequest(HurlStack.java:112)
        at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:97)
        at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:124)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "org.apache.http.ProtocolVersion" on path: DexPathList[[zip file "/data/app/org.babbageboole.binvenio-mK5w_R6fyrcGeElEdbF2Mw==/base.apk"],nativeLibraryDirectories=[/data/app/org.babbageboole.binvenio-mK5w_R6fyrcGeElEdbF2Mw==/lib/arm64, /system/lib64, /product/lib64]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:196)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at com.android.volley.toolbox.HurlStack.performRequest(HurlStack.java:112) 
        at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:97) 
        at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:124) 
```

**Comment by RobertBaruch** [comment]
Issue resolved.

When doing a search in Android Studio's "Add Library Dependency" search, "volley" came up with nothing useful, and "com.android.volley" showed v1.1.0 on JCenter, which causes the error. Using v1.1.1 as in the lesson fixes the issue.

**Comment by jpd236** [comment]
Glad 1.1.1 is working, but it's a bit unexpected that 1.1.0 wouldn't work because that was the release which should have made Apache HTTP optional. 1.0.0 on the other hand would be expected to see this error.

I'll leave this open to try to verify; if so, we should update https://github.com/google/volley/wiki/Migrating-from-Apache-HTTP to mention 1.1.1 as the requirement. Could you confirm 1.1.0 is broken but 1.1.1 is working with identical code?

**Comment by RobertBaruch** [comment]
I confirm. Note that what shows up in searching for com.android.volley using the dependency manager is:

- Group ID: com.android.volley
- Artifact Name: dc-volley
- Repository: JCenter
- Versions: only 1.1.0 listed.

Adding this leads to this line in build.gradle: `implementation 'com.android.volley:dc-volley:1.1.0'`

That is the version that fails.

In contrast, the version that succeeds wasn't found by the dependency manager, and is `implementation 'com.android.volley:volley:1.1.1'`.

I looked at the Maven repository, and found that [the correct version was there](https://mvnrepository.com/search?q=volley), but it was not in JCenter. I also searched the Google Maven repo, and it [wasn't there either](https://maven.google.com/web/index.html?q=volley). Since the Android Studio dependency manager looks at the Google and JCenter repos by default, the right version of volley wasn't found.

**Comment by jpd236** [comment]
Yes, com.android.volley:dc-volley appears to be a fork, and their [repository](https://github.com/DoveConviene/dc-volley) doesn't seem to have any of the ApacheHTTP related fixes. com.android.volley:volley:1.1.0 should work as well.

We publish Volley directly to [Bintray](https://bintray.com/android/android-utils/com.android.volley.volley/1.1.1#); I'm honestly not sure if/how it would propagate to other repositories. We can investigate that.

**Comment by RobertBaruch** [comment]
Nice find. OK, so root cause here is bintray -> Google/JCenter repo propagation isn't working. And I see you've changed the title. +1!

**Comment by paulsmithkc** [comment]
Does this mean that you don't need a particular private key, to sign modules for a given Group ID? That seems trivially exploitable if so...

**Comment by jpd236** [comment]
It seems that JCenter didn't do any sort of verification there, which seems problematic. On the other hand, JCenter is now deprecated and slated for shutdown, so there isn't much to be done about that now.

Google Maven only supports publishing of Google-owned libraries. Maven Central does their own verification that should prevent anyone not approved by a previous Google owner from publishing new artifacts underneath com.android (as we discovered when we began publishing to Maven Central recently).

So I think we might be able to close this out, as we publish to Google Maven and Maven Central directly, and this dc-volley fork is not in the com.android.volley namespace in either of those. I tested Android Studio's dependency search for "volley" and the first result shows our expected artifact under both Google and Maven Central.
